### PR TITLE
Threshold flow/v5

### PR DESCRIPTION
**Pull request by victorjulien**
SV_BRANCH=https://github.com/OISF/suricata-verify/pull/1691

Replaces #10582
- rebase
- add docs

https://redmine.openinfosecfoundation.org/issues/6822

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 19126

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 19134

File: src/detect-engine-register.h [hunk 1]
```diff
@@ -54,7 +54,18 @@ enum DetectKeywordId {
     DETECT_FLOW,
     /* end prefilter sort */
 
+    /* values used in util-var.c go here, to avoid int overflows
+     * TODO update var logic to use a larger type. */
```
Comment (lukashino) [inline src/detect-engine-register.h:58]: is this `todo` intended to be left here? a Redmine ticket seems more appropriate to me

**Comment by lukashino** [comment]
changes looks ok, the rebase is needed

**Comment by victorjulien** [comment]
replaced by #10632

**Pull request closed without merging**